core/src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaConsumerHelper.java

```diff
@@ -79,12 +79,18 @@ public static Consumer createConsumer(String bootStrapServers, String consumerPr
         }
     }
 
-    public static ConsumerRecords initialPollWaitingForConsumerGroupJoin(Consumer consumer) {
+    public static ConsumerRecords initialPollWaitingForConsumerGroupJoin(Consumer consumer, ConsumerCommonConfigs consumerCommonConfigs) {
+        // default wait time for poll
+        Long durationForPolling = 500L;
+        // use user defined consumer.pollingTime if provided
+        if (consumerCommonConfigs != null)
+            durationForPolling = ofNullable(consumerCommonConfigs.getPollingTime()).orElse(500L);
+        
         for (int run = 0; run < 10; run++) {
             if (!consumer.assignment().isEmpty()) {
                 return new ConsumerRecords(new HashMap());
             }
-            ConsumerRecords records = consumer.poll(Duration.of(500, ChronoUnit.MILLIS));
+            ConsumerRecords records = consumer.poll(Duration.of(durationForPolling, ChronoUnit.MILLIS));
```

Comment:
Looks good @dandalavinod . Can you add a test(couple of tests) to cover this?

It should be fairly easy if you follow the earlier PR related to this (https://github.com/authorjapps/zerocode/pull/422).
Probably you can duplicate the earlier test and amend it to setup a new scenario with this config param. That's it.

Reply:
@nirmalchandra thanks for the review. As you see, i have added/updated the current unit tests to support and verify fix. However, from integration level - i believe the current consume tests will verify my fix - since now for every consumer initial poll the code path i changed will be exercised and in turn be valdated ; as every consumer's consume call will tread through the intial poll join. Having said that - i added a integration test to verify case when the consumer.pollingTime is not provided in the properties file - please refer to my new commit related to these changes.